packages/medusa/src/api/routes/store/products/get-product.ts

```diff
@@ -134,15 +140,32 @@ export default async (req, res) => {
   const decoratePromises: Promise<any>[] = []
 
   if (shouldSetPricing) {
-    decoratePromises.push(
-      pricingService.setProductPrices([decoratedProduct], {
+    if (featureFlagRouter.isFeatureEnabled(PricingIntegrationFeatureFlag.key)) {
+      const context = await pricingService.collectPricingContext({
         cart_id: validated.cart_id,
         customer_id: customer_id,
         region_id: regionId,
         currency_code: currencyCode,
         include_discount_prices: true,
       })
-    )
+      decoratePromises.push(
+        getProductPricingWithPricingModule(
```

Comment:
Suggestion : instead of re fetching the variants, should we add a new argument to the product querying function to get the price_set at that moment. The util function would just take care of calculation. Same for the list product